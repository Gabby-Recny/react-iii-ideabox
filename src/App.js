import React, { Component } from 'react';
import Ideas from './Ideas';
import Form from './Form';
import { getIdeas, postIdea, deleteIdea } from './apiCalls.js';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      ideas: [],
      error: ''
    }
  }

  componentDidMount() {
    getIdeas()
      .then(ideas => this.setState({ ideas }))
      .catch(err => this.setState({ error: 'Something went wrong' }))
  }

  addIdea = (newIdea) => {
    postIdea(newIdea)
      .then(result => {
        if (result.id) {
          this.setState({ ideas: [...this.state.ideas, result], error: '' })
        } else {
          this.setState({ error: 'Please fill out both fields!' })
        }
      })
  }

  removeIdea = (id) => {
    deleteIdea(id)
      .then(response => {
        if (response.ok) {
          const filteredIdeas = this.state.ideas.filter(idea => idea.id !== id);
          
          this.setState({ ideas: filteredIdeas, error: '' });
        } else {
          this.setState({ error: `There was a problem deleting that idea!` })
        }
      })
  }

  render() {
    return(
      <main className='App'>
        <h1>IdeaBox</h1>
        <Form addIdea={this.addIdea} />
        {!!this.state.error &&
          <h2>{this.state.error}</h2>
        }
        {!this.state.error && !this.state.ideas.length &&
          <h2>Loading...</h2>
        }
        <Ideas ideas={this.state.ideas} removeIdea={this.removeIdea} />
      </main>
    )
  }
}

export default App;
