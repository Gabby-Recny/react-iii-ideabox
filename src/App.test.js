import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when addIdea is called', () => {
    const mockIdea = {
      id: 3, title: 'Sweaters for pugs', description: 'Why not?'
    };
    const expected = [{ id: 1, title: 'Prank Travis', description: 'Stick googly eyes on all his stuff' },
    { id: 2, title: 'Make a secret password app', description: 'So you and your rideshare driver can both know neither one of you is lying' }, mockIdea];

    wrapper.instance().addIdea(mockIdea);

    expect(wrapper.state('ideas')).toEqual(expected);
  });

  it('should update state when deleteIdea is called', () => {
    const idToDelete = 2;
    const expected = [{ id: 1, title: 'Prank Travis', description: 'Stick googly eyes on all his stuff' }];

    wrapper.instance().deleteIdea(idToDelete);

    expect(wrapper.state('ideas')).toEqual(expected);
  });
});
