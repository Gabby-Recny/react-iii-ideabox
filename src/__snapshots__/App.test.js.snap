// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<main
  className="App"
>
  <h1>
    IdeaBox
  </h1>
  <Form
    addIdea={[Function]}
  />
  <Ideas
    deleteIdea={[Function]}
    ideas={
      Array [
        Object {
          "description": "Stick googly eyes on all his stuff",
          "id": 1,
          "title": "Prank Travis",
        },
        Object {
          "description": "So you and your rideshare driver can both know neither one of you is lying",
          "id": 2,
          "title": "Make a secret password app",
        },
      ]
    }
  />
</main>
`;
